drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -377,6 +378,18 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 		return -EINVAL;
 	}
 
+	default_col = curr_dr_freq / m_rt->stream->params.rate / mstr_prop->default_row;
+	if (default_col != mstr_prop->default_col) {
+		/* Check if default_col can be changed */
+		list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
+			if (!s_rt->slave->id.class_id) {
+				dev_err(&s_rt->slave->dev, "The Peripheral doesn't comply with SDCA\n");
```

Comment:
you probably need a comment to explain that some devices expose the class ID but can't support dynamic scaling.

I would actually move this loop to a helper that  can be modified/quirked if needed, e.g. is_clock_scaling_supported().